### PR TITLE
Make GridGenerator::hyper_cube_with_cylindrical_hole<2>() more robust.

### PR DESCRIPTION
**Pull request by bangerth**
Chipping away at the test failures in #15689: This here is a function that makes specific assumptions about the vertex numbers of a triangulation. We do better by testing for the vertices' geometric location and save a few lines of code in the process.

### CODE REVIEW

**Review by drwells — APPROVED** [review]

**Review by tjhei — APPROVED** [review]

**Comment by tjhei** [comment]
```
The following tests FAILED:
	1866 - grid/grid_generator_plate_with_a_hole_4.debug (Failed)
``|

**Comment by bangerth** [comment]
This turned out to be a case where previously, a floating point number was considered equal than another and with my first commit it was ever so slightly smaller and failed the test. But the check was just poorly written. We specifically don't want to get into this round-off game, so now I added a tolerance and with that, the test succeeds.

While there, I also did a couple of other clean-ups in follow-up commits. The only relevant ones among the new ones is a3fe6f2fdb2d5c67b2c08c852452aa02f180e46c.

File: source/grid/grid_generator.cc [hunk 1]
```diff
@@ -3444,13 +3444,13 @@ namespace GridGenerator
...
           {
-            for (const unsigned int face_n : GeometryInfo<2>::face_indices())
+            for (const unsigned int face_n : cell->face_indices())
               {
                 const auto &face = cell->face(face_n);
                 if (face->at_boundary() &&
                     internal::point_in_2d_box(face->center(),
                                               center,
-                                              outer_radius))
+                                              outer_radius * (1. - 1e-12)))
```
Comment (bangerth) [inline source/grid/grid_generator.cc:3453]: For context: We have a domain that is a square on the outside and a circle on the inside, with 8 cells sitting in the gap. We want to attach a polar manifold to the circle on the inside, so we ask whether we are looking at a face that is (i) at the boundary, and (ii) on the *inside* of the box. This should only match the faces on the circle, but due to changes in round-off, one of the eight faces on the outside also matched the test, and we attached the wrong manifold to it. By using a slightly smaller box, we avoid this.

**Comment by bangerth** [comment]
Seems to work now!

**Pull request merged**